modules/sfp_peegeepee.py

```diff
@@ -0,0 +1,205 @@
+
+        time.sleep(1)
+
+        if res['content'] is None:
+            self.sf.debug('No response from d.PeeGeePee.com')
+            return None
+
+        pat = re.compile("(-----BEGIN.*END.*BLOCK-----)", re.MULTILINE | re.DOTALL)
+        matches = re.findall(pat, res['content'])
+
+        for match in matches:
+            self.sf.debug('Found public key: ' + match)
+
+            if len(match) < 300:
+                self.sf.debug('Likely invalid public key.')
+                continue
+
+            evt = SpiderFootEvent('PGP_KEY', match, self.__name__, sourceEvent)
+            self.notifyListeners(evt)
+
+        return None
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+
+        if srcModuleName == 'sfp_peegeepee':
+            return None
+
+        eventData = event.data
+        self.currentEventSrc = event
+
+        if eventData in self.results:
+            self.sf.debug("Skipping " + eventData + ", already checked.")
+            return None
+
+        self.results[eventData] = True
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        keys = self.query(eventData)
+
+        if not keys:
+            self.sf.debug('No results found for ' + eventData)
+            return None
+
+        # Get e-mail addresses on this domain
+        if eventName == 'DOMAIN_NAME' or eventName == 'INTERNET_NAME':
+            for key in keys:
+                email = keys.get(key)[1]
+
+                if not email:
+                    continue
+
+                mailDom = email.lower().split('@')[1]
+
+                if not self.getTarget().matches(mailDom):
+                    continue
+
+                self.sf.info('Found e-mail address: ' + email)
+                evt = SpiderFootEvent('EMAILADDR', email, self.__name__, event)
+                self.notifyListeners(evt)
+
+                # A bit of a hack. Submit the description to sfp_names
+                # and see if it is considered to be a name.
+                name = keys.get(key)[0]
+                evt = SpiderFootEvent('RAW_RIR_DATA', 'Possible full name: ' + name,
+                                      self.__name__, self.currentEventSrc)
+                self.notifyListeners(evt)
+
+                if self.opts['fetch_keys']:
+                    self.retrieveKey(key, event)
+ 
+        # Retrieve names for the specified e-mail address
+        if eventName == 'EMAILADDR':
+            for key in keys:
+                email = keys.get(key)[1]
+
+                if not email:
+                    continue
+
+                if not email.lower() == eventData.lower():
+                    continue
+
+                # A bit of a hack. Submit the description to sfp_names
+                # and see if it is considered to be a name.
+                name = keys.get(key)[0]
```

Comment:
This probably needs duplicate checking so that the same event for the same possible name isn't generated multiple times.

Reply:
Removed duplicates using a temporary `list()` uniqued with `set()`